cli/args/lockfile.rs

```diff
@@ -320,6 +523,54 @@ impl CliLockfile {
       lockfile: Mutex::new(lockfile),
       frozen: opts.frozen,
       skip_write: opts.skip_write,
+      use_lockfile_v5: opts.use_lockfile_v5,
+    })
+  }
+
+  pub fn read_from_path_current_version(
```

Comment:
should try to dedupe code between this and the async version above

Reply:
they're just different enough (and one is async) that it's hard to factor it out